sphinx/source/developers/consensus.rst

```diff
@@ -0,0 +1,41 @@
+Consensus Protocols
+===================
+
+In order to guarantee system reliability changes to the key-value store must be agreed by a number of nodes before being applied. The number of nodes that need to agree on a change before it is applied depends on the consensus protocol being run.
+
+The default consensus protocol for CCF is a version of Raft which provides Crash Fault Tolerance.
+
+We are currently working on providing PBFT as an additional consensus option. PBFT provides Byzantine Fault Tolerance.
+
+.. warning:: Currently CCF with PBFT is in development and should not be used in a production environment.
+
+Raft Consensus Protocol
+-----------------------
+
+We have implemented a version of Raft that provides Crash Fault Tolerance. This means that if we can tolerate up to ``f`` nodes crashing (or being un-responsive) then we need ``2*f + 1`` nodes in total to guarantee liveness. The majority of nodes that need to agree on a change is then ``f + 1``.
+Raft does not make any confidentiality or integrity claims if nodes have been compromised.
```

Comment:
I think this could be clearer - Raft _assumes that all nodes are honest_, and only protects against crashes (and crash-like behaviour) on these nodes. If _any_ are faulty or compromised, then confidentiality and integrity may be lost.

Reply:
Confidentiality should not be mentioned here, neither Raft nor PBFT have anything to do with it.

Reply:
I removed all confidentiality mentions